drift-idl-generator/src/main/java/com/facebook/drift/idl/generator/ThriftIdlGenerator.java

```diff
@@ -129,6 +131,46 @@ public ThriftIdlGenerator(ThriftIdlGeneratorConfig config, ClassLoader classLoad
 
         this.namespaces = config.getNamespaces();
         this.recursive = config.isRecursive();
+
+        for (String customCodecClassName : config.getCustomCodecs()) {
+            try {
+                Class<?> codecClass = load(customCodecClassName);
+                if (codecClass == null) {
+                    continue;
+                }
+                Object codec;
+
+                try {
+                    codec = codecClass.getDeclaredConstructor(ThriftCatalog.class).newInstance(catalog);
+                }
+                catch (NoSuchMethodException e) {
+                    try {
+                        codec = codecClass.getDeclaredConstructor().newInstance();
+                    }
+                    catch (NoSuchMethodException | InstantiationException | IllegalAccessException | InvocationTargetException nestedException) {
+                        throw new ThriftIdlGeneratorException("Can not find constructor for the custom codec.");
+                    }
+                }
+
+                if (codec instanceof ThriftCodec) {
+                    ThriftCodec<?> thriftCodec = (ThriftCodec<?>) codec;
+                    if (thriftCodec.getType() != null) {
+                        this.addCustomType(thriftCodec.getType());
+                        config.getVerboseLogger().accept("Registered custom codec: " + customCodecClassName);
+                    }
+                    else {
+                        config.getWarningLogger().accept("Codec " + customCodecClassName + " returned null type.");
```

Comment:
The method should not return null. This is a general implicit assumption in Presto codebase. Fail if method returns a `null`.